seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-common/src/main/java/org/apache/seatunnel/translation/spark/serialization/SeaTunnelRowConverter.java

```diff
@@ -178,9 +178,10 @@ public SeaTunnelRow unpack(GenericRow engineRow) throws IOException {
         SeaTunnelRowType rowType = (SeaTunnelRowType) dataType;
         RowKind rowKind = RowKind.fromByteValue(engineRow.getByte(0));
         String tableId = engineRow.getString(1);
-        Object[] fields = new Object[rowType.getTotalFields()];
-        for (int i = 0; i < fields.length; i++) {
-            fields[i] = reconvert(engineRow.get(i + 2), rowType.getFieldType(i));
+        Object[] fields = new Object[indexes.length];
+        for (int i = 0; i < indexes.length; i++) {
+            int fieldIndex = indexes[i];
```

Comment:
Fix mulit-table transform index mapping

cc @Carl-Zhou-CN

Reply:
You were right. It was my oversight

Reply:
How about add a unit test to cover it?

Reply:
added